### PR TITLE
Fix dead link and add CI check

### PR DESCRIPTION
**Pull request by CalvinKirs**
Descriptions of the changes in this PR:

### Motivation
I found some dead links while reading the README, to avoid this, I added a CI check.

### Changes
- Fix the dead links
- Add the markdown-link-check check

### CODE REVIEW

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
cool! 
you need to exclude the new file from the license check

**Comment by CalvinKirs** [comment]
> cool! you need to exclude the new file from the license check

Thanks for your reminder, I will exclude all JSON files, it's not necessary to check the license.

File: .github/workflows/dead-link-checker.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  pull_request:
+
+concurrency:
+  group: dlc-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  CheckDeadLinks:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
```
Comment (dlg99) [inline .github/workflows/dead-link-checker.yaml:29]: We only need to run this if select dirs have changes (e.g. `site3/`) and can skip for majority of the PRs

Comment (CalvinKirs) [reply .github/workflows/dead-link-checker.yaml:29]: Hi, I've thought of doing that, but if someone adds MD files to another dir, that may result in frequent maintenance of this directory (most times. people probably forget to do this).

It's a simple CI, so I'd rather run it for every PR, WDYT?

Comment (CalvinKirs) [reply .github/workflows/dead-link-checker.yaml:35]: now,the check is only triggered when the MD file is modified

Comment (CalvinKirs) [reply .github/workflows/dead-link-checker.yaml:35]: I didn't notice before that there is such a way to determine the file type 🤣

**Comment by CalvinKirs** [comment]
Good to go, PTAL, thx @dlg99 @nicoloboschi 
ps: I ignored some dead link checks because I'm not sure what the correct one should be, maybe we can create an issue to see if other familiar people are interested in doing it.
Another CI fail has nothing to do with this, maybe we need to rerun

File: .github/workflows/dead-link-checker.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    branches:
+      - master
+      - branch-*
+
+concurrency:
+  group: dlc-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  CheckDeadLinks:
```
Comment (nicoloboschi) [inline .github/workflows/dead-link-checker.yaml:33]: please use kebab-case (check-dead-links)

File: .github/workflows/dead-link-checker.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Dead Link Checker
+
+on:
+  push:
+    paths:
```
Comment (nicoloboschi) [inline .github/workflows/dead-link-checker.yaml:21]: this should be included also in the pull_request config?

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
really cool!

File: .github/workflows/dead-link-checker.yaml (hunk 2 above)
Comment (CalvinKirs) [reply .github/workflows/dead-link-checker.yaml]: Thx, done

File: .github/workflows/dead-link-checker.yaml (hunk 3 above)
Comment (CalvinKirs) [reply .github/workflows/dead-link-checker.yaml]: thanks you for pointing it out

**Review by nicoloboschi — APPROVED** [review]

**Comment by nicoloboschi** [comment]
rerun failure checks

**Review by hangc0276 — APPROVED** [review]

**Pull request merged**